### PR TITLE
[receiver/journald] Document permission requirements

### PR DESCRIPTION
**Pull request by FeltrinN**
**Description:**  Document permission requirements when running the collector as an installed package and docker container. Only minimal pointers for k8s provided as:
1. it requires further investigation
2. it should probably be implemented in detail within either the chart or the operator

**Link to tracking Issue:** #20906

**Testing:**  Running otelcol with the configuration provided in the issue and the permissions documented here results in log collection

**Documentation:** Documentation on required permissions when running as installed package and as container

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
Looks good, but I would add some command, so the user can quickly troubleshoot if the journaldreceiver should or shouldn't work on the system, eg:

```
su ${USERNAME} -c "journalctl --lines 5
sudo su ${USERNAME} -c "journalctl --lines 5
```

**Comment by FeltrinN** [comment]
> ```
> su ${USERNAME} -c "journalctl --lines 5
> sudo su ${USERNAME} -c "journalctl --lines 5
> ```

Added in a slightly different form because the user from the default package does not have a default shell (on ubuntu at least)... this should hopefully work everywhere :)

**Review by sumo-drosiek — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]
Thanks @FeltrinN!

File: receiver/journaldreceiver/README.md [hunk 1]
```diff
@@ -98,3 +101,34 @@ which is going to effectively retrieve all entries which matches the following s
   - `_SYSTEMD_UNIT` is `ssh`
   - `_SYSTEMD_UNIT` is `kubelet` and `_UID` is `1000`
 
+## Setup and deployment
+
+The user running the collector must have enough permissions to access the journal; not granting them will lead to issues.
+
+When running in a containerized enviroment, differences in the systemd version running on the host and on the container may prevent access to logs due to different features and configurations (e.g. zstd compression, keyed hash etc).
```
Comment (djaglowski) [inline receiver/journaldreceiver/README.md:108]: This should fix the lint failure: `"enviroment" is a misspelling of "environment"`
```suggestion
When running in a containerized environment, differences in the systemd version running on the host and on the container may prevent access to logs due to different features and configurations (e.g. zstd compression, keyed hash etc).
```

**Pull request merged**